### PR TITLE
54 add more stats to summary

### LINKED ISSUE #54: Add more stats to summary.html
- wallet balances of master node and slave nodes
- managedWallet balance
- A new table with tx volume / day

All wallet balances only for RBtc and with green/red background

### PR DESCRIPTION
**Pull request by Bitcoinera**
Closes #54

Missing a DB full of data to finish testing.

### CODE REVIEW

**Comment by Bitcoinera** [comment]
Looking really good

image

File: models/transaction.js [hunk 1]
```diff
@@ -60,10 +60,20 @@ export default class Transaction extends BaseModel {
     async countConfirmed(type) {
         try {
             const res = await this.get(`SELECT type, COUNT(*) FROM ${this.tableName} WHERE type = ? AND status = 'confirmed' GROUP BY type`, [type]);
-            return res || 0;
+            return res ? res['COUNT(*)'] : 0;
```
Comment (ztane) [inline models/transaction.js]: This is somewhat brittle though... depends on SQLite producing exactly that column name. I think it should just use `SELECT COUNT(*) AS ct FROM ...` in the query, and then `return res ? res.ct : 0;`

File: models/transaction.js [hunk 2]
```diff
@@ -60,10 +60,20 @@ export default class Transaction extends BaseModel {
...
         } catch (e) {
             console.error(e);
             return 0;
         }
     }
+
+    async countUnprocessed(type) {
+        try { 
+            const res = await this.get(`SELECT type, COUNT(*) FROM ${this.tableName} WHERE type = ? AND txId = NULL AND id > 100 AND status = 'confirmed' GROUP BY type`, [type]);
+            return res || 0;
```
Comment (ztane) [inline models/transaction.js]: This as well :D

**Comment by ztane** [comment]
Also, are those really BTC or satoshis?

**Comment by Bitcoinera** [comment]
> Also, are those really BTC or satoshis?

Yeah, well spotted. Turns out we were storing the transactions with their values in sats

**Comment by Bitcoinera** [comment]
Should be good now!

**Review by jamiemadrox779 — CHANGES_REQUESTED** [review]
dont want socket.io to be downgraded

**Comment by Bitcoinera** [comment]
> dont want socket.io to be downgraded

OK, fixed

**Pull request merged**